### PR TITLE
feat: backwards-compatible ctype data decoding

### PR DESCRIPTION
**Pull request by rflechtner**
Makes sure the sdk still decodes CType records from before the chain upgrade to 0.9.0 / 0.8.5.
Adjust tests so that CType fetching is skipped where not available.

## How to test:

We should no longer see the `Invalid empty address passed` errors in integration tests against `latest` (e.g., https://github.com/KILTprotocol/sdk-js/actions/runs/4003744234/jobs/6872161008).
~Fetching CType data will still fail with `Cannot fetch CType definitions on a chain that does not store the createdAt block`.~

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (ntn-x2) [inline packages/core/src/__integrationtests__/Ctypes.spec.ts:30]: So you're not using the storage version value anymore?

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (ntn-x2) [inline packages/core/src/__integrationtests__/Ctypes.spec.ts:34]: Version could be either 0 or 1 here, since 2 is the first valid one containing block number.

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (ntn-x2) [inline packages/core/src/__integrationtests__/Ctypes.spec.ts:33]: This `console.warn` triggers a lint warning. Is there a workaround? Maybe simply skipping it?

**Review by ntn-x2 — APPROVED** [review]
LGTM! The only thing that's not 100% clear is why we don't use the storage version number, that we decided to be 2 for the valid storage structure.

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:30]: I'm happy to use it once it's fixed!

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:33]: or a linter exception :D

Comment (ntn-x2) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:33]: Yeah I meant skipping the warning with a linter exception 😁

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (ntn-x2) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:30]: Ok, then as a placeholder, this is the PR that bumps the CType storage version to 2 for the new type: https://github.com/KILTprotocol/kilt-node/pull/458. Once it's merged into develop and an image is built, it can be used.

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:34]: but 0 is an invalid version number technically, if I got that right?

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:30]: Will put this on hold then until https://github.com/KILTprotocol/kilt-node/pull/458 is merged.

Comment (ntn-x2) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:30]: It has been. A new `latest-develop` tag is out.

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts:33]: actually that's something we can disable for all tests I think (https://github.com/KILTprotocol/sdk-js/pull/703/commits/ba18a01b1c480533470839794d5e6e53f0c7a000)

File: packages/core/src/__integrationtests__/Ctypes.spec.ts
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Ctypes.spec.ts]: https://github.com/KILTprotocol/sdk-js/pull/703/commits/6fd18eb4f7f104cb4b21dda2ca42eaf0304c17e0

**Pull request merged**
